FileShareService.DesktopClient.Core/Jobs/NewBatchJob.cs

```diff
@@ -52,11 +69,60 @@ public void Validate(JToken jsonToken)
             }
 
             //Check for files
-            if (jsonToken.SelectToken("actionParams.files") != null && 
+            if (jsonToken.SelectToken("actionParams.files") != null &&
                 jsonToken.SelectToken("actionParams.files").Type != JTokenType.Array)
             {
                 ErrorMessages.Add($"Invalid file object.");
             }
+
+            //Check for file attributes
+
+            foreach (JToken fileObj in jsonToken.SelectToken("actionParams.files"))
+            {
+                JToken? fileAttributeToken = fileObj.SelectToken("attributes");
+                var searchPath = fileObj.SelectToken("searchPath");
+
+                if (fileAttributeToken != null)
```

Comment:
I don't think i am being understood with my comments about reducing the indentation. Here is what i was thinking:

```
foreach (JToken fileObj in jsonToken.SelectToken("actionParams.files"))
{
    JToken? fileAttributeToken = fileObj.SelectToken("attributes");
    var searchPath = fileObj.SelectToken("searchPath");

    if (fileAttributeToken == null) continue;
    
    if (fileAttributeToken?.Type != JTokenType.Array)
    {
        ErrorMessages.Add("Invalid file attribute. searchPath:" + searchPath);
        continue;
    }
    
    if (!fileAttributeToken.HasValues) continue;
    
    //Check for file attribute key and value
    foreach (JToken? fileAttribute in fileAttributeToken)
    {
        JToken? fileKeyToken = fileAttribute.SelectToken("key");
        string fileKey = fileKeyToken?.Type == JTokenType.String ?
            Convert.ToString(fileKeyToken) : string.Empty;

        if (fileAttribute.SelectToken("key")?.Type != JTokenType.String)
        {
            ErrorMessages.Add($"File attribute key is missing or is invalid for the file. searchPath:" + searchPath);
        }
        else if (string.IsNullOrWhiteSpace(fileKey))
        {
            ErrorMessages.Add($"File attribute key cannot be blank. searchPath : " + searchPath);
        }


        JToken? fileValueToken = fileAttribute.SelectToken("value");
        string fileValue = fileValueToken?.Type == JTokenType.String ?
            Convert.ToString(fileValueToken) : string.Empty;
        if (fileAttribute.SelectToken("value")?.Type != JTokenType.String)
        {
            ErrorMessages.Add($"File attribute value is missing or is invalid for the file. searchPath : " + searchPath);
        }
        else if (string.IsNullOrWhiteSpace(fileValue))
        {
            ErrorMessages.Add($"File attribute value cannot be blank. searchPath : " + searchPath);
        }
    }
}
```

Reply:
updated the same. Thanks